src/codegen/codeparser.cpp

```diff
@@ -81,119 +86,157 @@ void CCodeParser::ParseCFiles(wxString className) {
+				m_userIncludeFilePreProc = code.substr(userIncludeStart, userIncludeEnd - userIncludeStart);
+			}
+		}
+	}
+}
+
+void CppCodeParser::ParseSourceFilePreProc(wxString &code) {
+	int userIncludeEnd;
+	m_userSourceFilePreProc.Clear();
+
+	// find the begining of the user include
+
+	int userIncludeStart = code.Find(wxT("//// end generated include. Place User Code Below"));
+	if (userIncludeStart != wxNOT_FOUND) 
+	{
+		userIncludeStart = code.find(wxT('\n'), userIncludeStart);
+		if (userIncludeStart != wxNOT_FOUND) 
+		{
+			// find the end of the user include
+			wxString Str = m_className + wxT("::");
+			userIncludeEnd = code.find(Str, userIncludeStart);
+			if (userIncludeEnd != wxNOT_FOUND) 
+			{
+				userIncludeEnd = code.rfind(wxT('\n'), userIncludeEnd);
+				if (userIncludeEnd != wxNOT_FOUND) 
+				{
+					userIncludeStart++;
+					m_userSourceFilePreProc = code.substr(userIncludeStart, userIncludeEnd - userIncludeStart);
+				}
 			}
 		}
 	}
 }
 
-void CCodeParser::ParseCClass(wxString code) {
+void CppCodeParser::ParseHeaderFileClass(wxString &code) {
 	int startClass = code.Find(wxT("class ") + m_className);
 	if (startClass != wxNOT_FOUND) {
 		code = ParseBrackets(code, startClass);
 		if (startClass != wxNOT_FOUND) {
-			ParseCUserMembers(code);
+			ParseHeaderFileUserMembers(code);
 		}
 	}
 }
 
-void CCodeParser::ParseCUserMembers(wxString code) {
-	m_userMemebers = wxT("");
-	int userMembersStart = code.Find(wxT("//// end generated class members"));
+void CppCodeParser::ParseHeaderFileUserMembers(wxString &code) {
+	m_userHeaderFileMembers = wxT("");
+	int userMembersStart = code.Find(wxT("//// end generated class members. Place User Code Below"));
 	if (userMembersStart != wxNOT_FOUND) {
 		userMembersStart = code.find('\n', userMembersStart);
 		if (userMembersStart == wxNOT_FOUND) {
-			m_userMemebers = wxT("");
+			m_userHeaderFileMembers = wxT("");
 		} else {
 			userMembersStart++;
 			if (userMembersStart < (int)code.Len()) {
-				m_userMemebers = code.Mid(userMembersStart);
+				m_userHeaderFileMembers = code.Mid(userMembersStart);
 			}
 		}
 	}
 }
 
-void CCodeParser::ParseSourceFunctions(wxString code) {
+void CppCodeParser::ParseSourceFileFunctions(wxString &code) {
 	int functionStart = 0;
 	int functionEnd = 0;
 	int previousFunctionEnd = 0;
 	wxString funcName, funcArg;
-	Function* func;
+	CppFunction* function;
 	wxString Str, R;
 
-	int loop = 0;
-	while (1) {
+	ParseSourceFilePreProc(code);
+
+	wxString codeSection;
+	while (functionStart != wxNOT_FOUND) {
 		// find the begining of the function name
 		Str = m_className + wxT("::");
 		functionStart = code.find(Str, previousFunctionEnd);
 		if (functionStart == wxNOT_FOUND) {
 			// Get the last bit of remaining code after the last function in the file
-			m_trailingCode = code.Mid(previousFunctionEnd);
-			m_trailingCode.RemoveLast();
+			if(code.GetChar(previousFunctionEnd) == '\n')
+				{previousFunctionEnd++;}
```

Comment:
This brace placement is bad, either use the Java style with brace on same line or place it on the next line, even for one-liners please.